### PR TITLE
add trivy scanner github action to report vulnarabiliites.

### PR DESCRIPTION
**Pull request by humblec**
This adds a trivy scanner test to report vulnerabilities.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
Thanks @guits ! looks like now we have the vulnarabilities reported on the base image here https://github.com/ceph/ceph-container/actions/runs/3470012183/jobs/5797688534

**Comment by mohag** [comment]
I don't think the Trivy check failing the pipelines will work - Trivy currently checks for RHEL vulnerabilities and still find some issues even with the latest CentOS Stream images currently. (possibly divergence of versioning?) (It might be useful to have it show up as warnings instead though)

File: .github/workflows/trivy.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: install go
+        uses: actions/setup-go@v3.3.1
+        with:
+          go-version: ^1.18
```
Comment (humblec) [inline .github/workflows/trivy.yaml:20]: this can be 1.20

File: .github/workflows/trivy.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW,UNKNOWN'
+
+      - name: run trivy vulnerability scanner against demo
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'ceph/demo:HEAD-main-centos-stream8-x86_64'
```
[the remark this answers is not in the document]
Comment (guits) [reply .github/workflows/trivy.yaml:39]: do you mean, a base OS (CentOS stream 8) up-to-date?

[the remark this answers is not in the document]
Comment (guits) [reply .github/workflows/trivy.yaml:39]: @humblec ^

[the remark this answers is not in the document]
Comment (humblec) [reply .github/workflows/trivy.yaml:39]: @guits yeah, that's what I meant , may be we can leave it for now.

**Pull request closed without merging**